### PR TITLE
[Selectable Components: Structured List - Usage] Website Doc

### LINKED ISSUE #3862: [Selectable Components: Structured List - Usage] Website Doc
Closes [#744](https://github.com/carbon-design-system/carbon-design-kit/issues/744#issue-2064434013)

### Update the Structured List website (Usage tab) documentation with what came out of our [final design solutions](https://github.com/carbon-design-system/carbon/issues/14766).

----

### USAGE TAB

**Default preselection content**
> - [x] Change our guidance to not mention we have structured list preselected by default. This occurs in the [Interaction section](https://arc.net/l/quote/yrkdofsf) for selectable tile and check to see if it is anywhere else.

**Anchor links**
> - [x] Add "Default", "Selectable" and "Modiifers" links 
> - [x] Bring it into 2 rows

**_Overview_ section**
> - [x] Change the headings "When to use" and "When not to use" from H4 to H3.

**_Variant_ section**
> - [x] Add the callout section to inform about the experimental structures list (feature flag)
> - [x] Add link to Default & Selectable in _Variant_ section table 

**_Alignment_ section**
> - [x] Add this line - "We do not offer the flush alignment with selectable functionality."

**_States_ section**
> - [x] Add 6 "States" (H4) section along with description, tables : Enabled, Hover, Focus, Selected, Disabled, Skeleton.

**_Interaction_ section**
> - [x] Update the copy under "Selectable structured list", update this line 
> 1. "By default, no option will be preselected."
> 2. "When the user selects an item from the list, the selected row will appear with the `radio-button--checked` icon."

Add 3 sections (all H2) along with images (with real product images), description -
> - [x] **_Default_ section** 
> - [x] **_Selectable_ section** 
> - [x] **_Modifiers_ section** 

**Selectable structured list images**
> - [x] Correct any images with the updated visual styling.

---

- [x] PR Review -> Merge

### PR DESCRIPTION
**Pull request by Kritvi-bhatia17**
Closes #3862

For ref: [Figma file](https://www.figma.com/file/w4QA4bnNZAiySPskA5qZMW/Selectable-Components?type=design&node-id=1068-38426&mode=design&t=Ui7SEG60RG1bG4pH-4)

---

#### Changelog

**_Variant_** section
> - [x] Added the inline notification for experimental structured list 
> - [x] Added links to the Variant table

> - [x] Add a new section(**_States_** section) with description and table
> - [x] Add a few new sections with description and images each: **_Default_**, **_Selectable_**  & **_Modifiers_** section

---

### Changed

**_Interactions_** section
> - [x] Updated the copy

### CODE REVIEW

File: src/pages/components/structured-list/usage.mdx [hunk 1]
```diff
@@ -138,16 +162,86 @@ alignment.
...
+
+Default structured lists are used to display information to the user, such as
+pricing plans. These structured list can have internal calls-to-action (CTAs),
+such as a [button](/components/button/usage) or
+a [link](/components/link/usage).
+
+<Row>
+<Column colLg={12}>
+
+![default structured list](images/structured-list-usage-4.png)
```
Comment (laurenmrice) [inline src/pages/components/structured-list/usage.mdx:203]: Just some typos:

- In the left most column, the word "Related" has a typo.
- There is an extra space in the first paragraph between "Director" and "platform".
- There is an extra space in the first paragraph between "hosting," and "operations,".
- There is an extra space in the first paragraph between "can" and "quickly".
- Under the "Single-tenant" column in the structured list, the first sentence should say the word "require" not the word required.
- Under the "Multi-tenant" column in the structured list,  in the second row, the beginning of the sentence needs to be capitalized and be "On-demand".

File: src/pages/components/structured-list/usage.mdx [hunk 2]
```diff
@@ -138,16 +162,86 @@ alignment.
...
+Selectable structured lists work well for presenting options to a user in a
+structured manner, such as a set of pricing plans. Selectable tiles may contain
+internal CTAs (like links to docs) if the internal CTA is given its own click
+target. Selectable structured lists have a single-select state working like
+a [radio button](/components/radio-button/usage).
+
+<Row>
+<Column colLg={12}>
+
+![selectable structured list](images/structured-list-usage-5.png)
```
Comment (laurenmrice) [inline src/pages/components/structured-list/usage.mdx:221]: I made some comments in your Figma file about some spacing in one of the rows in the structured list in this image.

There are also a few typos: 
- In the first left column there is a typo in the word "Category".
- In the right panel, there is a typo in the word "Washington".
- In the right panel, in the terms, the word should be "agreed" and not agree.

The radio icons are also not aligned correctly.

File: src/pages/components/structured-list/usage.mdx [hunk 3]
```diff
@@ -138,16 +162,86 @@ alignment.
...
+<Caption> Selectable structured list </Caption>
+
+</Column>
+</Row>
+
+## Modifiers
+
+### Background
+
+Background has been added to the structured list, and it is optional. The
```
Comment (laurenmrice) [inline src/pages/components/structured-list/usage.mdx:232]:
```suggestion
A background has been added to the structured list, and it is optional. The
```

File: src/pages/components/structured-list/usage.mdx [hunk 4]
```diff
@@ -138,16 +162,86 @@ alignment.
...
+background is only present for the hang alignment structured list, both default
```
Comment (laurenmrice) [inline src/pages/components/structured-list/usage.mdx:233]:
```suggestion
background is only available for the hang alignment structured list in default
```

File: src/pages/components/structured-list/usage.mdx [hunk 5]
```diff
@@ -138,16 +162,86 @@ alignment.
...
+and selectable; it is not included for flush alignment.
```
Comment (laurenmrice) [inline src/pages/components/structured-list/usage.mdx:234]:
```suggestion
and selectable variants. It is not available for the flush alignment structured list.
```

**Review by laurenmrice — CHANGES_REQUESTED** [review]
Looking good Kritvi! :tada: Just a couple things:

File: src/pages/components/structured-list/usage.mdx [hunk 6]
```diff
@@ -88,7 +108,7 @@ Structured list is composed of two sections—column header and data row.
 <Row>
 <Column colLg={12}>
 
-![Structured list anatomy](images/structured-list-usage-1.png)
+![structured list anatomy](images/structured-list-usage-1.png)
```
[the remark this answers is not in the document]
Comment (Kritvi-bhatia17) [reply src/pages/components/structured-list/usage.mdx:111]: I've made the picture as 522 px.

File: src/pages/components/structured-list/usage.mdx [hunk 7]
```diff
@@ -138,16 +162,86 @@ alignment.
...
+| _Focus_    | When a user tabs or clicks on a row, it becomes focused, indicating the user has successfully navigated to the component (except default structured list).                                                                        |
+| _Selected_ | When a user presses space or clicks on a structured list row to select it.                                                                                                                                                        |
+| _Disabled_ | When the user is not allowed to interact with a selectable structured list due to either permissions, dependencies, or pre-requisites.                                                                                            |
+| _Skeleton_ | When the user can review but not modify the component. This state removes all interactive functions like the disabled state but can still be focusable, accessible by screen readers, and passes visual contrast for readability. |
```
[the remark this answers is not in the document]
Comment (laurenmrice) [reply src/pages/components/structured-list/usage.mdx:180]: This seems to still need to be changed @Kritvi-bhatia17

File: src/pages/components/structured-list/usage.mdx [hunk 8]
```diff
@@ -138,16 +162,86 @@ alignment.
...
 - If you need to select multiple items, use a
   [data table.](/components/data-table/usage)
 - When the user selects an item from the list, the selected row will appear with
-  the `checkmark--filled` icon.
+  the `radio-button--checked` icon.
+
+## Default
+
+Default structured lists are used to display information to the user, such as
+pricing plans. These structured list can have internal calls-to-action (CTAs),
```
[the remark this answers is not in the document]
Comment (laurenmrice) [reply src/pages/components/structured-list/usage.mdx:196]: This still needs to be updated

File: src/pages/components/structured-list/usage.mdx (hunk 2 above)
Comment (laurenmrice) [reply src/pages/components/structured-list/usage.mdx:221]: There is still a typo in the word "Washington" in the right panel.

File: src/pages/components/structured-list/usage.mdx (hunk 7 above)
[the remark this answers is not in the document]
Comment (Kritvi-bhatia17) [reply src/pages/components/structured-list/usage.mdx:180]: Ohh got it!

File: src/pages/components/structured-list/usage.mdx (hunk 8 above)
[the remark this answers is not in the document]
Comment (Kritvi-bhatia17) [reply src/pages/components/structured-list/usage.mdx:196]: Oh shit sorry I guess I missed it, done!

File: src/pages/components/structured-list/usage.mdx (hunk 2 above)
Comment (Kritvi-bhatia17) [reply src/pages/components/structured-list/usage.mdx:221]: Ohh sorry, yes done!

**Review by alisonjoseph — APPROVED** [review]

**Review by laurenmrice — APPROVED** [review]
Looks good! Just realized we usually bold the states in the States section so I suggested the change.

**Review by tw15egan — APPROVED** [review]

**Pull request merged**